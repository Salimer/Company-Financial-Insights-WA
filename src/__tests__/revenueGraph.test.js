import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import RevenueGraph from '../components/revenueGraph';

jest.mock('../functions/getCompanyName', () => jest.fn().mockReturnValue('Company Name'));

describe('RevenueGraph', () => {
  test('renders correctly', () => {
    const company = [
      { symbol: 'ABC', date: '2021-01-01' },
      { symbol: 'ABC', date: '2022-01-01' },
      { symbol: 'ABC', date: '2023-01-01' },
    ];
    const { container } = render(
      <MemoryRouter>
        <RevenueGraph company={company} />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
