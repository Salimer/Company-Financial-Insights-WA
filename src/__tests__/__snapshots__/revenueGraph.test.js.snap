// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RevenueGraph renders correctly 1`] = `
<div>
  <section
    class="sc-bgqQcB bJDtsW grid grid-cols-2 w-full h-screen"
  >
    <a
      class="bg-buttonFirstColor p-5 text-white flex flex-col justify-center items-center border-white border"
      href="/undefined/2021"
    >
      <span>
        <ion-icon
          name="calendar-outline"
          size="large"
        />
      </span>
      <span
        class=" text-lg tracking-wider font-bold"
      >
        2021
      </span>
    </a>
    <a
      class="bg-buttonFirstColor p-5 text-white flex flex-col justify-center items-center border-white border"
      href="/undefined/2022"
    >
      <span>
        <ion-icon
          name="calendar-outline"
          size="large"
        />
      </span>
      <span
        class=" text-lg tracking-wider font-bold"
      >
        2022
      </span>
    </a>
    <a
      class="bg-buttonFirstColor p-5 text-white flex flex-col justify-center items-center border-white border"
      href="/undefined/2023"
    >
      <span>
        <ion-icon
          name="calendar-outline"
          size="large"
        />
      </span>
      <span
        class=" text-lg tracking-wider font-bold"
      >
        2023
      </span>
    </a>
  </section>
</div>
`;
